lib/ruby-enum/enum/ecase.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module Ruby
+  module Enum
+    ##
+    # Adds a method to an enum class that allows for exhaustive matching on a value.
+    #
+    # @example
+    # class Color
+    #   include Ruby::Enum
+    #   include Ruby::Enum::Ecase
+    #
+    #   define :RED, :red
+    #   define :GREEN, :green
+    #   define :BLUE, :blue
+    # end
+    #
+    # Color.ecase(Color::RED, {
+    #   [Color::RED, Color::GREEN] => -> { puts "red or green" },
+    #   Color::BLUE => -> { puts "blue" },
+    # })
+    #
+    # Reserves the :else key for a default case:
+    # Color.ecase(Color::RED, {
+    #   [Color::RED, Color::GREEN] => -> { puts "red or green" },
+    #   else: -> { puts "blue" },
+    # })
+    module Ecase
```

Comment:
Ruby doesn't have enums, so we add `Enum`, but ruby does have `case`. Does calling this `Ruby::Enum::Case`, a special kind of case, not work? Will there be ever a different `Xcase`? Finally, should it be `ECase` vs. `Ecase`?

Reply:
Very good point! It's way nicer to have `Ruby::Enum::Case` and `Color.case` 👍